src/convertToObject.js

```diff
@@ -1,12 +1,38 @@
 'use strict';
 
 /**
- * @param {string} sourceString
+ * Converts CSS string to object.
  *
- * @return {object}
+ * @param {string} sourceString CSS code.
+ *
+ * @return {Object} CSS code converted to keys and values.
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return (
+    sourceString
+      // Handle cases where the string might start with whitespace or semicolons
+      .trim()
+      // Breaks sourceString by semicolon into separate stings
+      .split(';')
+
+      // Breaks line into property and value and removes extra space charachters
+      .map((line) => line.split(':').map((el) => el.trim()))
+
+      // Leaves only valid elements
+      .filter(
+        (el) =>
+          el.length === 2 &&
+          // Robust check that both elements are not empty strings
+          el.every((element) => element !== ''),
```

Comment:
You don't need to compare with an empty string, just check if elements exists https://developer.mozilla.org/en-US/docs/Glossary/Truthy

Reply:
It was suggested by ai-mentor :robot: https://github.com/mate-academy/js_style-to-object/pull/3380#discussion_r1675958870